### PR TITLE
HttpConnectionPool doPOST overloaded for handling formParams

### PR DESCRIPTION
**Pull request by jyotigupta**

### CODE REVIEW

File: http-handler/src/main/java/com/flipkart/poseidon/handlers/http/impl/HttpConnectionPool.java [hunk 1]
```diff
@@ -16,12 +16,33 @@
...
+import org.apache.http.HttpRequest;
+import org.apache.http.HttpRequestInterceptor;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpResponseInterceptor;
+import org.apache.http.NameValuePair;
 import org.apache.http.client.HttpClient;
 import org.apache.http.client.entity.GzipCompressingEntity;
 import org.apache.http.client.entity.GzipDecompressingEntity;
-import org.apache.http.client.methods.*;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
```
Comment (shrey-garg) [inline http-handler/src/main/java/com/flipkart/poseidon/handlers/http/impl/HttpConnectionPool.java]: please revert the import changes

Comment (jyotigupta) [inline http-handler/src/main/java/com/flipkart/poseidon/handlers/http/impl/HttpConnectionPool.java]: imports are better understood with '*' that is why I had done that explicitly.
http://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad

anyhow, there is no problem in changing back, will do that :)

File: http-handler/src/main/java/com/flipkart/poseidon/handlers/http/impl/HttpConnectionPool.java [hunk 2]
```diff
@@ -231,6 +248,16 @@ public HttpResponse doPOST(String uri, byte[] data, Map<String, String> requestH
         setRequestHeaders(request, requestHeaders);
         return execute(request);
     }
+    
+    /**
+     * Method for executing HTTP POST request with form params
+     */
+    public HttpResponse doPOST(String uri, List<NameValuePair> formParams , Map<String, String> requestHeaders) throws Exception {
+        HttpPost request = new HttpPost(constructUrl(uri));
+        request.setEntity(new UrlEncodedFormEntity(formParams));
```
Comment (pmohankumar) [inline http-handler/src/main/java/com/flipkart/poseidon/handlers/http/impl/HttpConnectionPool.java]: For completeness, we need to support compressing form params as well. Something like
if (this.requestGzipEnabled) {
                request.addHeader(CONTENT_ENCODING, COMPRESSION_TYPE);
                request.setEntity(new GzipCompressingEntity(new UrlEncodedFormEntity(formParams)));
            } else {
                request.setEntity(new UrlEncodedFormEntity(formParams));
}

UrlEncodedFormEntity changes Content-Type header (to "application/x-www-form-urlencoded") and requestGzipEnabled controls Content-Encoding header.

Comment (jyotigupta) [inline http-handler/src/main/java/com/flipkart/poseidon/handlers/http/impl/HttpConnectionPool.java]: Thanks 👍  updating the CR

**Comment by jyotigupta** [comment]
Closing this in favour of https://github.com/flipkart-incubator/Poseidon/pull/103

**Pull request closed without merging**